src/view/com/composer/char-progress/CharProgress.tsx

```diff
@@ -27,35 +26,51 @@ export function CharProgress({
   const pal = usePalette('default')
   const textColor = count > maxLength ? '#e60000' : pal.colors.text
   const circleColor = count > maxLength ? '#e60000' : pal.colors.link
+  const remainingCount = maxLength - count
+  const shouldShowText = remainingCount <= 20
+  const shouldShowCircle = remainingCount > -10
+
   return (
     <View
-      style={[a.flex_row, a.align_center, a.justify_between, a.gap_sm, style]}>
-      <Text
-        style={[
-          {color: textColor, fontVariant: ['tabular-nums']},
-          a.flex_grow,
-          a.text_right,
-          textStyle,
-        ]}>
-        {maxLength - count}
-      </Text>
-      {count > maxLength ? (
-        <ProgressPie
-          size={size ?? 30}
-          borderWidth={4}
-          borderColor={circleColor}
-          color={circleColor}
-          progress={Math.min((count - maxLength) / maxLength, 1)}
-        />
-      ) : (
-        <ProgressCircle
-          size={size ?? 30}
-          borderWidth={1}
-          borderColor={pal.colors.border}
-          color={circleColor}
-          progress={count / maxLength}
-        />
+      style={[
+        a.flex,
+        a.align_center,
+        a.justify_center,
+        a.gap_sm,
+        styles.container,
```

Comment:
Hi! New here so I'm not familiar with the overall styling but imo since you're bothering to make a styles object, move the flex styles into that too just to declutter the HTML a bit

Reply:
I figured we could just reuse globally available styles as much as possible. I am also not sure what is the best approach here to comply with Bluesky guidelines. I will wait until someone from a core team jumps in and makes recommendations/suggestions here.